controllers/dashboard_test.go

```diff
@@ -30,17 +33,153 @@ var _ = Describe("DashboardController", func() {
 	})
 
 	AfterEach(func() {
+		TruncateTable("keyword")
 		TruncateTable("user")
 	})
 
 	Describe("GET /dashboard", func() {
 		Context("when the user has already signed in", func() {
 			It("renders with status 200", func() {
-				user := FabricateUser("dev@nimblehq.co", "password")
+				user := FabricateUser(faker.Email(), faker.Password())
 				response := MakeAuthenticatedRequest("GET", "/dashboard", nil, user)
 
 				Expect(response.StatusCode).To(Equal(http.StatusOK))
 			})
+
+			Context("given 12 records of the keyword in the database", func() {
+				It("shows 2 pages in pagination", func() {
+					pageClass := "pagination__page"
+
+					totalRecords := 12
+					user := FabricateUser(faker.Email(), faker.Password())
+					for i := 1; i <= totalRecords; i++ {
+						_ = FabricateKeyword(fmt.Sprintf("key%02dword", i), "https://www.sample.com", user)
+					}
+
+					response := MakeAuthenticatedRequest("GET", "/dashboard", nil, user)
+					body, err := ioutil.ReadAll(response.Body)
+					if err != nil {
+						Fail("Read body content failed: " + err.Error())
+					}
+					err = response.Body.Close()
+					if err != nil {
+						Fail("Close body content failed: " + err.Error())
+					}
+
+					r, err := regexp.Compile(pageClass)
+					if err != nil {
+						Fail("Regexp failed: " + err.Error())
+					}
+
+					matches := r.FindAllString(string(body), -1)
+
+					Expect(len(matches)).To(BeNumerically("==", 3))
```

Comment:
> shows 2 pages in pagination

Should it be matched with `3`?

Reply:
@junan Because it's including with navigation link.
I've update it to the other selector with this `data-page-number`, so it will check just only real page.